drivers/hopwa_caper/app/models/hopwa_caper/generators/fy2024/enrollment_filters/strmu_longevity_filter.rb

```diff
@@ -0,0 +1,59 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HopwaCaper::Generators::Fy2024::EnrollmentFilters
+  # STRMU is one-time service. Expect a new enrollment/household every time service is provided
+  StrmuLongevityFilter = Struct.new(:label, :client_ids, keyword_init: true) do
+    def apply(scope)
+      scope.where(destination_client_id: client_ids)
+    end
+
+    def self.for_report(report)
+      program_filter = HopwaCaper::Generators::Fy2024::EnrollmentFilters::ProjectFunderFilter.strmu_hopwa
+
+      rows = program_filter.apply(report.hopwa_caper_enrollments).
+        where(hopwa_eligible: true).
+        where(entry_date: (report.start_date - 5.years)..report.end_date).
+        group(:destination_client_id).
+        pluck(:destination_client_id, Arel.sql('ARRAY_AGG(entry_date)'))
+
+      # since the report asks us to sum these, assume they are expected mutually exclusive
+      buckets = {
+        'How many households have been served by STRMU for the first time this year?' => [],
+        'How many households also received STRMU assistance during the previous year?' => [],
+        'How many households received STRMU assistance more than twice during the previous five years?' => [],
+        'How many households received STRMU assistance during the last five consecutive years?' => [],
+      }
+
+      rows.each do |client_id, entry_dates|
+        # what years are covered by prior enrollments
+        entry_years = entry_dates.map(&:year).uniq.select
+        bucket = nil
+        if entry_years.size >= 5
+          bucket = 'How many households received STRMU assistance during the last five consecutive years?'
```

Comment:
Do we care that this says consecutive, but that entry dates might not be?  
Does each client get a new enrollment every year?  
What happens if a client has a single enrollment that lasts 5 years?

Reply:
Good questions

> Do we care that this says consecutive, but that entry dates might not be?

since we are only looking at the enrollments for last five years, if there are five unique years then they're likely contiguous; I'm sure there are cases where this doesn't hold up though. I'll make it more percise

> Does each client get a new enrollment every year? What happens if a client has a single enrollment that lasts 5 years?

It sounded like strmu enrollments tend to be short-lived. Perhaps we shouldn't assume that though and account for the exit date

Reply:
If we're looking at 5 years of data and counting 5 distinct years, that feels fine for "contiguous for 5 years" I was thinking we were looking at the 15 year look back and just counting that they had 5, which wasn't feeling right.